src/wx/CMakeLists.txt

```diff
@@ -38,488 +168,235 @@ if(NOT WIN32 AND NOT APPLE)
+# Direct3D.
+if(NOT ENABLE_DIRECT3D)
+    target_compile_definitions(visualboyadvance-m PRIVATE NO_D3D)
+endif()
 
-            set(CMAKE_REQUIRED_DEFINITIONS ${CURRENT_DEFS})
-            set(CMAKE_REQUIRED_LIBRARIES   ${CURRENT_LIBS})
-        endif()
+# FAudio.
+if(ENABLE_FAUDIO)
+    find_package(FAudio REQUIRED)
+    target_link_libraries(visualboyadvance-m FAudio::FAudio)
+else()
+    target_compile_definitions(visualboyadvance-m PRIVATE NO_FAUDIO)
+endif()
 
-        if(WX_ABI_FOUND_MATCH)
-            # add C++ flags
-            if(CMAKE_COMPILER_IS_GNUCXX)
-                string(REGEX REPLACE "<FLAGS>" "<FLAGS> -fabi-version=${WX_ABI_VERSION} " CMAKE_CXX_COMPILE_OBJECT "${CMAKE_CXX_COMPILE_OBJECT}")
-                set(CMAKE_CXX_LINK_EXECUTABLE "${CMAKE_CXX_LINK_EXECUTABLE} -fabi-version=${WX_ABI_VERSION}")
-            elseif(CMAKE_CXX_COMPILER_ID STREQUAL Clang)
-                string(REGEX REPLACE "<FLAGS>" "<FLAGS> -D__GXX_ABI_VERSION=10${WX_ABI_VERSION} " CMAKE_CXX_COMPILE_OBJECT ${CMAKE_CXX_COMPILE_OBJECT})
-                set(CMAKE_CXX_LINK_EXECUTABLE "${CMAKE_CXX_LINK_EXECUTABLE} -D__GXX_ABI_VERSION=10${WX_ABI_VERSION}")
-            endif()
-        endif()
-    endif()
+# wxWidgets.
+set(wxWidgets_USE_UNICODE ON)
+if(CMAKE_BUILD_TYPE MATCHES "^(Debug|RelWithDebInfo)$")
+    set(wxWidgets_USE_DEBUG ON) # noop if wx is compiled with --disable-debug, like in Mac Homebrew atm
+endif()
+if(VBAM_STATIC)
+    set(wxWidgets_USE_STATIC ON)
+endif()
+find_package(wxWidgets COMPONENTS xrc xml html adv net core base gl)
+if(NOT wxWidgets_FOUND)
+    set(WX_HAS_OPENGL FALSE)
+    find_package(wxWidgets COMPONENTS xrc xml html adv net core base REQUIRED)
+endif()
 
-    # now check for OpenGL
-    include(CheckCXXSourceCompiles)
+target_link_libraries(visualboyadvance-m ${wxWidgets_LIBRARIES})
+target_include_directories(visualboyadvance-m PRIVATE ${wxWidgets_INCLUDE_DIRS})
+target_compile_options(visualboyadvance-m PRIVATE ${wxWidgets_CXX_FLAGS})
+target_compile_definitions(visualboyadvance-m PRIVATE ${wxWidgets_DEFINITIONS})
+if(CMAKE_BUILD_TYPE MATCHES "^(Debug|RelWithDebInfo)$")
+    target_compile_definitions(visualboyadvance-m PRIVATE ${wxWidgets_DEFINITIONS_DEBUG})
+endif()
 
-    if(NOT DEFINED WX_HAS_OPENGL)
-        check_cxx_source_compiles("
+list(APPEND CMAKE_REQUIRED_LIBRARIES ${wxWidgets_LIBRARIES})
+list(APPEND CMAKE_REQUIRED_INCLUDES ${wxWidgets_INCLUDE_DIRS})
+list(APPEND CMAKE_REQUIRED_FLAGS ${wxWidgets_CXX_FLAGS})
+foreach(DEF ${wxWidgets_DEFINITIONS})
+    list(APPEND CMAKE_REQUIRED_DEFINITIONS "-D${DEF}")
+endforeach()
+
+# Check for OpenGL.
+include(CheckCXXSourceCompiles)
+if(NOT DEFINED WX_HAS_OPENGL)
+    check_cxx_source_compiles("
 #include <wx/wx.h>
 #include <wx/config.h>
 #include <wx/glcanvas.h>
 
 int main(int argc, char** argv) {
-    wxGLCanvas canvas(NULL, wxID_ANY, NULL, wxPoint(0, 0), wxSize(300, 300), 0);
-    return 0;
+wxGLCanvas canvas(NULL, wxID_ANY, NULL, wxPoint(0, 0), wxSize(300, 300), 0);
```

Comment:
We could get rid of this too, the gl `COMPONENTS` check should be sufficient. I don't remember why I did this exactly, I think the `COMPONENTS` check wasn't working, but all of this likely works fine now.

Reply:
Thanks, done!